### PR TITLE
Jda v5

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
Update to v5
Fixed a lot issues such as now were member can be null

### CODE REVIEW

File: jda5/build.gradle [hunk 1]
```diff
@@ -25,10 +25,10 @@ repositories {
...
     implementation project(':annotations')
-    implementation 'org.junit.jupiter:junit-jupiter:5.8.1'
+    implementation 'org.junit.jupiter:junit-jupiter:5.8.2'
 
     //JDA
-    implementation group: 'net.dv8tion', name: 'JDA', version: '4.4.0_350'
+    implementation group: 'net.dv8tion', name: 'JDA', version: '5.0.0-alpha.2'
     // https://mvnrepository.com/artifact/ch.qos.logback/logback-classic
     implementation group: 'ch.qos.logback', name: 'logback-classic', version: '1.2.5'
     implementation group: 'com.sedmelluq', name: 'lavaplayer', version: '1.3.78'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply jda5/build.gradle:34]: ignore

File: jda.basic/build.gradle [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+dependencies {
+    implementation 'org.junit.jupiter:junit-jupiter:5.8.2'
+
+    //JDA
+    implementation group: 'net.dv8tion', name: 'JDA', version: '4.4.0_351'
+    implementation group: 'io.github.yusufsdiscordbot', name: 'annotations', version: '1.0.3'
+    // https://mvnrepository.com/artifact/ch.qos.logback/logback-classic
+    implementation group: 'ch.qos.logback', name: 'logback-classic', version: '1.2.5'
+    implementation group: 'com.sedmelluq', name: 'lavaplayer', version: '1.3.78'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply jda.basic/build.gradle:34]: ignore

File: javacord/build.gradle [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        name 'm2-dv8tion'
+        url 'https://m2.dv8tion.net/releases'
+    }
+}
+
+dependencies {
+    //JDA
+    implementation group: 'org.javacord', name: 'javacord', version: '3.3.2'
+    //Log4j Core
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.11.0'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord/build.gradle:30]: ignore

File: javacord/build.gradle [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    implementation group: 'com.sedmelluq', name: 'lavaplayer', version: '1.3.77'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord/build.gradle:31]: ignore

File: javacord.basic/build.gradle [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        url 'https://m2.dv8tion.net/releases'
+    }
+}
+
+dependencies {
+    //JDA
+    implementation group: 'org.javacord', name: 'javacord', version: '3.3.2'
+    //Log4j Core
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.11.0'
+    implementation group: 'com.sedmelluq', name: 'lavaplayer', version: '1.3.77'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord.basic/build.gradle:31]: ignore

File: javacord.basic/build.gradle [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        name 'm2-dv8tion'
+        url 'https://m2.dv8tion.net/releases'
+    }
+}
+
+dependencies {
+    //JDA
+    implementation group: 'org.javacord', name: 'javacord', version: '3.3.2'
+    //Log4j Core
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.11.0'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord.basic/build.gradle:30]: ignore

File: application.basic/build.gradle [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+dependencies {
+    implementation 'org.junit.jupiter:junit-jupiter:5.8.2'
+
+    //JDA
+    implementation group: 'net.dv8tion', name: 'JDA', version: '4.4.0_351'
+    implementation group: 'io.github.yusufsdiscordbot', name: 'annotations', version: '1.0.3'
+    // https://mvnrepository.com/artifact/ch.qos.logback/logback-classic
+    implementation group: 'ch.qos.logback', name: 'logback-classic', version: '1.2.5'
+    implementation group: 'com.sedmelluq', name: 'lavaplayer', version: '1.3.78'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply application.basic/build.gradle]: ignore

File: javacord.basic/build.gradle (hunk 5 above)
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord.basic/build.gradle:31]: ignore

File: javacord.basic/build.gradle [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    mavenCentral()
+    maven {
+        name 'm2-dv8tion'
+        url 'https://m2.dv8tion.net/releases'
+    }
+}
+
+dependencies {
+    //JDA
+    implementation group: 'org.javacord', name: 'javacord', version: '3.3.2'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord.basic/build.gradle:28]: ignore

File: javacord/build.gradle [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    mavenCentral()
+    maven {
+        name 'm2-dv8tion'
+        url 'https://m2.dv8tion.net/releases'
+    }
+}
+
+dependencies {
+    //JDA
+    implementation group: 'org.javacord', name: 'javacord', version: '3.3.2'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord/build.gradle]: ignore

File: javacord/build.gradle (hunk 4 above)
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply javacord/build.gradle]: ignore

File: jda5/src/main/java/io/github/yusufsdiscordbot/yusufsdiscordcore/bot/slash_command/interactions/Command.java [hunk 10]
```diff
@@ -80,9 +80,6 @@ protected Command(String name, String description, boolean isGuildOnly) {
      *         Choices can also be used which makes it easier for the user. which returns
      *         {@link OptionData#addChoice(String, long)} <br>
      *         <br>
-     *         Original from <a href=
-     *         "https://github.com/Together-Java/TJ-Bot/blob/95d7f323a998b15abfa2c0723c30636d2f00c4cf/application/src/main/java/org/togetherjava/tjbot/commands/SlashCommand.java#L91">here</a>
-     *         and modified by Yusuf
```
Comment (Zabuzard) [inline jda5/src/main/java/io/github/yusufsdiscordbot/yusufsdiscordcore/bot/slash_command/interactions/Command.java]: Add it back.

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: jda5/src/main/java/io/github/yusufsdiscordbot/yusufsdiscordcore/bot/slash_command/interactions/Command.java (hunk 10 above)
Comment (RealYusufIsmail) [reply jda5/src/main/java/io/github/yusufsdiscordbot/yusufsdiscordcore/bot/slash_command/interactions/Command.java]: Its at the top

Comment (Zabuzard) [reply jda5/src/main/java/io/github/yusufsdiscordbot/yusufsdiscordcore/bot/slash_command/interactions/Command.java]: Okay.

**Pull request merged**